### PR TITLE
Linear Adapter 5: Workflow lifecycle and handoff mapping

### LINKED ISSUE #69: Linear Adapter 5: Workflow Lifecycle And Handoff Mapping
## Goal

Map Linear workflow states and ticket-driven handoffs into Symphony's internal handoff model.

## Scope

Use the Elixir workflow as the primary reference for Linear-specific lifecycle semantics:

- `Todo`
- `In Progress`
- `Human Review`
- `Rework`
- `Merging`
- `Done`
- configured active / terminal state sets

This issue should define how the Linear adapter interprets those states as normalized internal handoff outcomes for the orchestrator.

## Constraints

1. Keep Linear workflow semantics in the adapter/policy layer.
2. Do not make the orchestrator branch directly on Linear state names.
3. Preserve the repo-owned plan review and PR follow-up concepts while mapping them onto a Linear-centered tracker workflow.
4. Keep the design recoverable: a fresh factory should be able to infer the relevant handoff state from Linear plus repo state.

## Deliverables

- Linear lifecycle policy module
- mapping from configured Linear states into internal handoff states
- tests for `Human Review`, `Rework`, `Merging`, active, and terminal cases
- explicit design notes on how the Linear ticket conversation/workpad participates in recovery and handoff

## Acceptance

The adapter can express Linear ticket workflow semantics through the tracker-neutral internal handoff model without forcing the orchestrator to know Linear state names.

### PR DESCRIPTION
**Pull request by jessmartin**
## Summary
- map Linear workflow states and review markers into tracker-neutral handoff lifecycle outcomes
- move successful Linear runs into `Human Review` and keep terminal issues recoverable until Symphony records completion
- add unit, integration, and e2e coverage for Human Review, Rework, Merging, active, terminal, and recovery behavior

Closes #69

### CODE REVIEW

**Comment by jessmartin** [comment]
Addressed the Linear policy follow-up in `ca10766`.

- made configured `active_states` and `terminal_states` match case-insensitively across classification, handoff inspection, and state updates
- changed same-timestamp review-signal resolution to preserve original comment order instead of UUID lexical order
- added unit and integration coverage for both edge cases

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was started, but the CLI stalled after diff inspection without returning findings; no findings were emitted before the stall.

**Comment by jessmartin** [comment]
Addressed the remaining Linear review follow-up in `d54c031`.

- fixed the mixed-case duplicate adjacent active-state no-op in the Linear claim policy
- extracted the shared Linear state-name comparator into `src/tracker/linear-state-name.ts`
- skipped redundant terminal-state writes in `completeIssue`
- added a warning and regression coverage for projects that do not expose `Human Review`

Local checks after the final patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was run again on the final diff. It inspected the diff and relevant source files, then stalled without emitting findings or exiting, matching the earlier CLI behavior.

File: src/tracker/linear.ts [hunk 1]
```diff
@@ -142,17 +143,28 @@ export class LinearTracker implements Tracker {
...
+    await this.#writer.updateIssue(
+      {
+        id: issue.id,
+        description: updatedDescription,
+        ...(humanReviewStateName === null ||
+        humanReviewStateName === issue.state.name
+          ? {}
+          : { stateName: humanReviewStateName }),
+      },
+      project,
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply src/tracker/linear.ts:184]: Addressed in d54c031. `reconcileSuccessfulRun` now emits a warning when the Linear project does not expose `Human Review`, and `tests/integration/linear.test.ts` covers that degraded path while preserving the tracker-neutral handoff result.

File: src/tracker/linear-policy.ts [hunk 2]
```diff
@@ -69,8 +88,9 @@ export function resolveLinearTerminalStateName(
   config: LinearTrackerConfig,
 ): string {
   for (const stateName of config.terminalStates) {
-    if (project.states.some((state) => state.name === stateName)) {
-      return stateName;
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply src/tracker/linear-policy.ts:73]: Fixed in d54c031. The duplicate-adjacent guard now reuses the shared case-insensitive Linear state-name comparator, and `tests/unit/linear-policy.test.ts` covers the mixed-case duplicate case (`["Todo", "in progress", "IN PROGRESS"]`).

File: src/tracker/linear-write.ts [hunk 3]
```diff
@@ -93,3 +95,7 @@ function requireProjectForStateUpdate(
     `Linear issue update for ${issueId} requires project workflow state lookup`,
   );
 }
+
+function sameStateName(left: string, right: string): boolean {
+  return left.localeCompare(right, undefined, { sensitivity: "accent" }) === 0;
+}
```
[the remark this answers is not in the document]
Comment (jessmartin) [reply src/tracker/linear-write.ts]: Fixed in d54c031. The duplicated comparator was extracted to `src/tracker/linear-state-name.ts`, and both `linear-policy.ts` and `linear-write.ts` import the shared helper now.

**Comment by jessmartin** [comment]
Addressed the remaining Greptile follow-up in `23808d1`.

- kept `reconcileSuccessfulRun()` from moving issues backward out of Linear review workflow states by treating `Human Review`, `Rework`, and `Merging` as already-in-review
- added integration coverage for the `Rework` and `Merging` regression cases
- documented the legacy dotted `"Plan ready for review."` marker in both review-signal parsers and added a unit test that the undotted variant stays invalid

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was run again. Like the earlier attempts on this PR branch, the CLI inspected the diff and related files but stalled without emitting findings or a terminal result.

**Comment by jessmartin** [comment]
Addressed the latest Cursor follow-up in `f9814b3`.

- preserved configured terminal Linear states during `reconcileSuccessfulRun()`, so externally-completed issues do not get moved back to `Human Review`
- added an integration regression test for the `Done` case
- extracted the shared repo-owned plan review marker parser into `src/tracker/plan-review-signal.ts` so Linear and plan-review policy stay aligned on the same signal set

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was run again and repeated the same prior behavior: it inspected the diff and related files but stalled without returning findings or a terminal exit.

**Comment by jessmartin** [comment]
Addressed the latest Greptile follow-up in `44e2618`.

- skipped `HANDOFF_READY_COMMENT` when a successful run lands on an issue already in `Rework` or `Merging`, while still allowing the normal Human Review handoff path
- added regression assertions that `reconcileSuccessfulRun()` leaves those states untouched and does not post the misleading handoff-ready comment
- documented and unit-tested the intended precedence where review workflow states stay `running` even if the workpad still says `failed`

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was run again on this branch, but the CLI timed out after diff inspection without emitting findings.

**Comment by jessmartin** [comment]
Addressed the remaining Cursor follow-up in `3a8e8dd`.

- documented the intended precedence that a configured terminal Linear workflow state is the primary recovery signal, even if the workpad still carries a stale `failed` status
- added a unit regression test that terminal-state issues with stale failed workpads stay in the running/recovery set until Symphony writes the final `completed` workpad

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was rerun on the final diff and again timed out after diff inspection without emitting findings.

**Comment by jessmartin** [comment]
Addressed the latest Cursor/Greptile follow-up in `e43e081`.

- made `isLinearHumanReviewWorkflowState()` private because it is only used inside `linear-policy.ts`
- added unit coverage for the previously untested `waived` and bare `handoff-ready` branches in `createLinearHandoffLifecycle()`
- stopped posting duplicate handoff-ready comments when `reconcileSuccessfulRun()` is called on an issue already in `Human Review`, while preserving the normal first transition into review
- added an integration regression test for the already-in-`Human Review` case

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was rerun on the refreshed diff and again timed out after diff inspection without emitting findings.

**Comment by jessmartin** [comment]
Addressed the remaining Cursor follow-up in `ea52303`.

- made `isLinearReworkWorkflowState()` and `isLinearMergingWorkflowState()` private because they are only used inside `linear-policy.ts`
- kept the public Linear lifecycle surface unchanged apart from removing those accidental exports

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test -- tests/unit/linear-policy.test.ts` (this invocation still exercised the full Vitest suite in this repo, and it passed)

`codex review --base origin/main` was rerun on the updated branch. Like the prior attempts on this PR, it inspected the diff and relevant source files, then stalled without emitting findings or a terminal result.

**Comment by jessmartin** [comment]
Addressed the remaining Greptile follow-up in `38ceb94`.

- reused the `LinearIssueWriter.updateIssue()` snapshot inside `reconcileSuccessfulRun()` instead of immediately re-fetching the same issue through `inspectIssueHandoff()`
- added an integration regression assertion that a claim plus successful-run reconcile only performs the expected two `GetProjectIssue` reads total
- kept the lifecycle decision on the same tracker-policy path, but removed the extra network round-trip and its write/read race window

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was run again on this branch. Like the prior attempts on PR #80, it inspected the diff and related source files, then stalled without emitting findings or a terminal result; I interrupted it after confirming no findings had been produced.

**Comment by jessmartin** [comment]
Addressed the stale Linear review-signal follow-up in `7820666`.

- fenced Linear review markers to the current handoff cycle by ignoring approval/rework comments that predate the active `handoff-ready` workpad entry
- added a unit regression test that stale approved comments no longer move a fresh `Human Review` handoff straight to `awaiting-system-checks`
- added an integration regression test for `reconcileSuccessfulRun()` with a pre-existing approved comment from an older run

Local checks after the patch:
- `pnpm format:check`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`

`codex review --base origin/main` was run again. It repeated the prior behavior on this branch: it inspected the diff and invoked `git diff`/`git show`, but never emitted findings or a terminal result before stalling.
